api/query/cache/index/index_test.go

```diff
@@ -148,5 +151,146 @@ func TestEvictNonEmpty(t *testing.T) {
+					Status: "PASS",
+				},
+			},
+		}, nil
+	}).AnyTimes()
+
+	// Baseline before running things in parallel: Index already contains 8 runs.
+	i.IngestRun(makeRun(1))
+	i.IngestRun(makeRun(2))
+	i.IngestRun(makeRun(3))
+	i.IngestRun(makeRun(4))
+	i.IngestRun(makeRun(5))
+	i.IngestRun(makeRun(6))
+	i.IngestRun(makeRun(7))
+	i.IngestRun(makeRun(8))
+
+	// Eight times (from run IDs 9 through 16), in parallel:
+	// - Evict one run,
+	// - Add one run,
+	// - Attempt one query (that may fail to bind if it references an already
+	//   already evicted run).
+	var wg sync.WaitGroup
+	for j := 9; j <= 16; j++ {
+		wg.Add(1)
+		go func(n int) {
+			defer wg.Done()
+			i.EvictAnyRun()
+		}(j)
+		wg.Add(1)
+		go func(id int64) {
+			defer wg.Done()
+			i.IngestRun(makeRun(id))
+		}(int64(j))
+		wg.Add(1)
+		go func(n int) {
+			defer wg.Done()
+			plan, err := i.Bind([]shared.TestRun{
+				makeRun(int64(n - 1)),
+				makeRun(int64(n - 2)),
+				makeRun(int64(n - 3)),
+				makeRun(int64(n - 4)),
+			}, query.TestStatusConstraint{
+				BrowserName: "Chrome",
+				Status:      shared.TestStatusPass,
+			})
+			if err != nil {
+				return
+			}
+
+			plan.Execute()
+		}(j)
+	}
+	wg.Wait()
+
+	// Number of runs should now be 8 + 8 - 8 = 8.
+	// Shards, taken together, should contain data for two predictable run results
+	// for each run still in the index. (See loader.EXPECT()...DoAndReturn(...)
+	// callback above for predictable test names and values.)
+
+	// TODO: Should Index have a Runs() getter for purposes such as this check?
+	idx, ok := i.(*shardedWPTIndex)
+	assert.True(t, ok)
+
+	assert.Equal(t, 8, len(idx.runs))
+	sharedTestID, err := computeTestID("shared", nil)
+	assert.Nil(t, err)
+	numResults := 0
+	for _, s := range idx.shards {
+		// TODO: Should Results have a getter for purposes such as this check?
+		results, ok := s.results.(*resultsMap)
+		assert.True(t, ok)
+		numRuns := 0
+		results.byRunTest.Range(func(key, value interface{}) bool {
+			numRuns++
+			return true
+		})
+		assert.Equal(t, 8, numRuns)
+
+		for _, run := range idx.runs {
+			value, ok := results.byRunTest.Load(RunID(run.ID))
+			assert.True(t, ok)
+			// TODO: Should Results have a getter for purposes such as this check?
+			res, ok := value.(*runResultsMap)
+			assert.True(t, ok)
+
+			strID := strconv.FormatInt(run.ID, 10)
+			expectedTestID, err := computeTestID("test"+strID, nil)
+			assert.Nil(t, err)
+
+			for testID, resultID := range res.byTest {
+				// Either test is the "shared test" with varied result values across
+				// runs or it is the "test-specific test" with name `test<test ID>` and
+				// result value of "PASS".
+				assert.True(t, sharedTestID == testID || (expectedTestID == testID && resultID == ResultID(shared.TestStatusPass)))
+				numResults++
+			}
+		}
+	}
+
+	// Total number of results is 8 runs * 2 results per run = 16.
+	assert.Equal(t, 16, numResults)
+}
+
+var browsers = []string{
+	"Chrome",
+	"Edge",
+	"Firefox",
+	"Safari",
```

Comment:
These names should technically be lower-case, but it doesn't really matter in this test.